frontend/src/core/user/components/FileUpload.js

```diff
@@ -0,0 +1,70 @@
+/* @flow */
+
+import * as React from 'react';
+import { Localized } from 'fluent-react';
+
+import './FileUpload.css';
+
+import type { NavigationParams } from 'core/navigation';
+
+
+type Props = {|
+    parameters: NavigationParams,
+|};
+
+type Ref = {
+    current: React.ElementRef<any> | null,
+};
+
+/*
+ * Render a File Upload button.
+ */
+export default class FileUpload extends React.Component<Props> {
+    uploadForm: Ref;
+
+    constructor(props: Props) {
+        super(props);
+        this.uploadForm = React.createRef();
+    }
+
+    submitForm = () => {
+        const current = this.uploadForm.current;
+        if (current) {
+            current.submit();
+        }
+    }
+
+    render() {
+        const { parameters } = this.props;
+
+        let csrfToken = '';
+        const rootElt = document.getElementById('root');
+        if (rootElt) {
+            csrfToken = rootElt.dataset.csrfToken;
+        }
```

Comment:
I wasn't able to figure out how to re-use `getCSRFToken` from `core.api.base` here. Help?

Reply:
Added a comment.